src/NHibernate.Test/SystemTransactions/SystemTransactionFixture.cs

```diff
@@ -643,6 +650,152 @@ public void CanUseSessionWithManyDependentTransaction(bool explicitFlush)
 			}
 		}
 
+		// This test check a concurrency issue hard to reproduce. If it is flaky, it has to be considered failing.
+		// In such case, raise triesCount to investigate it locally with more chances of triggering the trouble.
+		[Test]
+		public void SupportsTransactionTimeout()
+		{
+			Assume.That(TestDialect.SupportsTransactionScopeTimeouts, Is.True, "The tested dialect is not supported for transaction scope timeouts.");
+			// Other special cases: ODBC and SAP SQL Anywhere succeed this test only with transaction.ignore_session_synchronization_failures
+			// enabled.
+			// They freeze the session during the transaction cancellation. To avoid the test to be very long, the synchronization
+			// lock timeout should be lowered too.
+
+			// A concurrency issue exists with the legacy setting allowing to use the session from transaction completion, which
+			// may cause session leaks. Ignore them.
+			FailOnNotClosedSession = !UseConnectionOnSystemTransactionPrepare;
```

Comment:
The PostgreSQL CI has again failed with the session leak. I am adding a condition to ignore that leak.